src/core/instance/render-helpers/render-list.js

```diff
@@ -26,10 +26,25 @@ export function renderList (
     }
   } else if (isObject(val)) {
     keys = Object.keys(val)
-    ret = new Array(keys.length)
-    for (i = 0, l = keys.length; i < l; i++) {
-      key = keys[i]
-      ret[i] = render(val[key], key, i)
+    if (keys.length === 0 && val.toString().indexOf('Iterator') > -1) {
```

Comment:
`Iterator` looks like new to me. Would you like to give me a link of introduction to `Iterator`? I might be wrong but Google only gave me `Generator`.

Reply:
1.  [Map.prototype.keys()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/keys) =>return a new Map iterator object.   
2. [Set.prototype.values()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/values) => return a new Iterator object containing the values for each element in the given Set, in insertion order.

Reply:
[Iteration_protocols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols)

Reply:
Thanks. I have checked the spec and MapIterator's `@@toStringTag` symbol returns `Map Iterator`. The similar for other iterators.

GIven this, I suspect if a `toString` test is legitimate. Checking `Symbol.iterator` in proper environment is better, I think.

Reply:
Yes, the better way is checking Symbol.iterator. At first I thought I can't use the ES6 grammar.

Reply:
Symbol.toStringTag
A string value used for the default description of an object. Used by Object.prototype.toString().
[Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)

Reply:
Symbol.toStringTag
A string value used for the default description of an object. Used by Object.prototype.toString().
[Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)

Reply:
`Symbol.toStringTag`
A string value used for the default description of an object. Used by `Object.prototype.toString()`.
[Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)